src/register/mcause.rs

```diff
@@ -52,13 +93,10 @@ impl Interrupt {
     #[inline]
     pub fn from(nr: usize) -> Self {
         match nr {
```

Comment:
Can you just give the nums values?
```
pub enum Interrupt {
    SupervisorSoft = 1,
    MachineSoft = 3,
...
``` 
Then when you want the integer value do `Interrupt::SupervisorSoft as u32`

Reply:
This is for a `usize` -> `Interrupt` conversion. There is no core support for doing that as a cast, but I could use [num-derive](https://docs.rs/num-derive/latest/num_derive/) to automatically generate methods to cast between integers. I didn't do this initially as I'm a bit wary of adding new dependencies for this sort of crate, but I think in this case it may be justified, as having casts available to any integer size would be nice.